### ISSUE TITLE
Can not override .default(r.error(....)) in `Model.get`

### ISSUE THREAD

**Issue by soplakanets** [neumino/thinky#299]
Currently, when I do a query, like `User.get(someId)`
I get an error because of the following lines in `lib/query.js`:

```
      case 'get':
        // `get` in thinky returns an error if the document is not found.
        // The driver currently just returns `null`.
        (function(key) {
          Query.prototype[key] = function() {
            return new Query(this._model, this._query[key].apply(this._query, arguments)).default(this._r.error(new Errors.DocumentNotFound().message));
          }
        })(key);
        break;
```

What was the reason for implementing it like this? I would like to at least be able to do `User.get(someId).default(null)` which is not possible in current implementation.

**Comment by neumino** [comment]
For:
```
Model.get(someId).default(null).run()
```
If it was working, it would get rejected if the returned value is null since you can't build an object with null.

**Comment by soplakanets** [comment]
But I could use .execute() or set something meaningful for default in such case. My point is, it  breaks the consistency and is quite unexpected. What about `Model.getAll(id1,id2).run()`? Should it also set .default(r.error()) ?

**Comment by neumino** [comment]
`Model.getAll(id1,id2).run()` returns an empty sequence if nothing matches, in which case you can do `merge` on it for example. It will work as it will loop over the element to merge, and since there are none, it's basically a no op.

Doing `Model.get(...).merge(...)` typically will break if `get` returns null.

**Comment by soplakanets** [comment]
@neumino I understand the reasoning I would like to at least be able to override the default. Currently it is not possible.

I would like to be able to to do Model.get(id).default(null).execute() and manually check the result for null.

**Comment by neumino** [comment]
@soplakanets -- I just released 2.0.11. You can use `_get` if you want `get` without a `default`. This should let give you all the flexibility you need :)

**Comment by soplakanets** [comment]
@neumino Thanks. It'll do a job too.
